creator-node/src/routes/files.js

```diff
@@ -168,286 +171,172 @@ const logGetCIDDecisionTree = (decisionTree, req) => {
-    } catch (e) {
-      decisionTree.push({
-        stage: `COMPUTE_LEGACY_FILE_PATH_FAILURE`
-      })
-      logGetCIDDecisionTree(decisionTree, req)
-      return sendResponse(
-        req,
-        res,
-        errorResponseBadRequest(`${logPrefix} Invalid CID`)
-      )
-    }
-
-    /**
-     * Check if file exists on FS at legacyStoragePath
-     * If found and not of file type, return error
-     * If found and of type file, continue with legacyStoragePath
-     * If not found, continue
-     */
-    startMs = Date.now()
     try {
-      // Will throw if path does not exist
-      // If exists, returns an instance of fs.stats class
-      fsStats = await fs.stat(legacyStoragePath)
-      decisionTree.push({
-        stage: `FS_STATS_LEGACY_STORAGE_PATH`,
-        time: `${Date.now() - startMs}ms`
-      })
-
+      /**
+       * fs.stat returns instance of fs.stats class, used to check if exists, an dir vs file
+       * Throws error if nothing found
+       * https://nodejs.org/api/fs.html#fspromisesstatpath-options
+       */
+      fsStats = await fs.stat(currStoragePath)
       if (fsStats.isFile()) {
-        decisionTree.push({
-          stage: `CID_CONFIRMED_FILE_LEGACY_STORAGE_PATH`
-        })
         if (CID !== EMPTY_FILE_CID && fsStats.size === 0) {
           // Remove file if it is empty and force fetch from CN network
-          await fs.unlink(storagePath)
-          decisionTree.push({
-            stage: `EMPTY_FILE_FOUND_AND_REMOVED_LEGACY_PATH`
-          })
+          await fs.unlink(currStoragePath)
         } else {
           fileFoundOnFS = true
+          storagePath = currStoragePath
+
+          mode = currPathType
+          break
         }
       } else if (fsStats.isDirectory()) {
-        decisionTree.push({
-          stage: `CID_CONFIRMED_DIRECTORY_LEGACY_STORAGE_PATH`
-        })
-        logGetCIDDecisionTree(decisionTree, req)
-        return sendResponse(
+        return sendResponseWithMetric(
           req,
           res,
-          errorResponseBadRequest('this dag node is a directory')
+          errorResponseBadRequest('this dag node is a directory'),
+          { result: 'abort_cid_is_directory', mode: currPathType }
         )
       } else {
-        decisionTree.push({
-          stage: `CID_INVALID_TYPE_LEGACY_STORAGE_PATH`
-        })
-        logGetCIDDecisionTree(decisionTree, req)
-        return sendResponse(
+        return sendResponseWithMetric(
           req,
           res,
-          errorResponseBadRequest('CID is of invalid file type')
+          errorResponseBadRequest('CID is of invalid file type'),
+          { result: 'abort_cid_invalid_type', mode: currPathType }
         )
       }
     } catch (e) {
-      decisionTree.push({
-        stage: `FS_STATS_LEGACY_STORAGE_PATH_CID_NOT_FOUND`,
-        time: `${Date.now() - startMs}ms`
-      })
-      // continue
-    }
-
-    if (fileFoundOnFS) {
-      storagePath = legacyStoragePath
+      // swallow error
```

Comment:
Would it be useful to record the error somehow. 

e.g. `tracing.recordException(e)`